### ISSUE TITLE
How to add new parition to an existing topic 

### ISSUE THREAD

**Issue by Mdhaker** [SOHU-Co/kafka-node#1224]
Hello,

is it possible to do an alter on an existing topic to add a partition like this  ? : 
`bin/kafka-topics.sh --zookeeper zk_host:port/chroot --alter --topic my_topic_name 
   --partitions 40 ` 
` AdminUtils.createOrUpdateTopicPartitionAssignmentPathInZK(topic, partitionReplicaList, zkClient, true)
`

**Comment by hyperlink** [comment]
Are you asking how altering topic partitions affects kafka-node? ConsumerGroup has a schedule to recognize when new partitions are added and will rejoin the group to consume the new partition.

**Comment by Mdhaker** [comment]
@hyperlink  No, I am asking if there is a feature in kafka-node to let me do that ? I want to add paritions programmatically in my case, thats why I m looking for altering !

**Comment by hyperlink** [comment]
Unfortunately, that API is not implemented.

**Comment by jlandersen** [comment]
Part of the protocol and can be done without going to zookeeper (https://kafka.apache.org/protocol#The_Messages_CreatePartitions).

I'll be happy to take a look at it.
